python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/ecs.py

```diff
@@ -0,0 +1,250 @@
+        cluster: Optional[str] = None,
+        group: Optional[str] = None,
+        network_configuration: Optional[Dict[str, Any]] = None,
+        overrides: Optional[Dict[str, Any]] = None,
+        placement_constraints: Optional[List[Dict[str, Any]]] = None,
+        placement_strategy: Optional[List[Dict[str, Any]]] = None,
+        platform_version: Optional[str] = None,
+        reference_id: Optional[str] = None,
+        started_by: Optional[str] = None,
+        tags: Optional[List[Dict[str, str]]] = None,
+        client_token: Optional[str] = None,
+        volume_configurations: Optional[List[Dict[str, Any]]] = None,
+        capacity_provider_strategy: Optional[List[Dict[str, Any]]] = None,
+        enable_ecs_managed_tags: Optional[bool] = None,
+        enable_execute_command: Optional[bool] = None,
+        propagate_tags: Optional[Literal["TASK_DEFINITION", "SERVICE", "NONE"]] = None,
+    ) -> PipesClientCompletedInvocation:
+        """Start a ECS task, enriched with the pipes protocol.
+
+        See also: `AWS API Documentation <https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RunTask.html>`_
+
+        Returns:
+            PipesClientCompletedInvocation: Wrapper containing results reported by the external
+            process.
+        """
+        with open_pipes_session(
+            context=context,
+            message_reader=self._message_reader,
+            context_injector=self._context_injector,
+            extras=extras,
+        ) as session:
+            pipes_args = session.get_bootstrap_env_vars()
+
+            overrides = overrides or {}
+            overrides["containerOverrides"] = overrides.get("containerOverrides", [])
+
+            # get all containers from task definition
+            task_definition_response = self._client.describe_task_definition(
+                taskDefinition=task_definition
+            )
+
+            log_configurations = {
+                container["name"]: container.get("logConfiguration")
+                for container in task_definition_response["taskDefinition"]["containerDefinitions"]
+            }
+
+            all_container_names = {
+                container["name"]
+                for container in task_definition_response["taskDefinition"]["containerDefinitions"]
+            }
+
+            container_names_with_overrides = {
+                container_override["name"] for container_override in overrides["containerOverrides"]
+            }
+
+            if isinstance(self._context_injector, PipesEnvContextInjector):
+                # set env variables for every container in the taskDefinition
+                # respecting current overrides provided by the user
+
+                environment_overrides = [
+                    {
+                        "name": k,
+                        "value": v,
+                    }
+                    for k, v in pipes_args.items()
+                ]
```

Comment:
1) Let's call `pipes_args = session.get_bootstrap_env_vars()` closer to its use
2) This should be called with _any_ context injector, not just `PipesEnvContextInjector`. Even with other injectors the "bootstrap" env vars are always injected.. For example if passing context via s3, the s3 bucket name would be in the bootstrap arg.

Reply:
oh that's right, will fix